state-chain/pallets/cf-lending-pools/src/lib.rs

```diff
@@ -0,0 +1,973 @@
+
+			for update in updates {
+				match update {
+					PalletConfigUpdate::SetNetworkFeeDeductionFromBoost { deduction_percent } =>
+						NetworkFeeDeductionFromBoostPercent::<T>::set(deduction_percent),
+				}
+				Self::deposit_event(Event::<T>::PalletConfigUpdated { update });
+			}
+
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::add_boost_funds())]
+		pub fn add_boost_funds(
+			origin: OriginFor<T>,
+			asset: Asset,
+			amount: AssetAmount,
+			pool_tier: BoostPoolTier,
+		) -> DispatchResult {
+			let booster_id = T::AccountRoleRegistry::ensure_liquidity_provider(origin)?;
+
+			ensure!(T::SafeMode::get().add_boost_funds_enabled, Error::<T>::AddBoostFundsDisabled);
+
+			ensure!(amount > Zero::zero(), Error::<T>::AddBoostAmountMustBeNonZero);
+
+			// `try_debit_account` does not account for any unswept open positions, so we sweep to
+			// ensure we have the funds in our free balance before attempting to debit the account.
+			T::PoolApi::sweep(&booster_id)?;
+
+			T::Balance::try_debit_account(&booster_id, asset, amount)?;
+
+			let boost_pool: BoostPool =
+				BoostPools::<T>::get(asset, pool_tier).ok_or(Error::<T>::BoostPoolDoesNotExist)?;
+
+			let core_pool_id = boost_pool.core_pool_id;
+
+			CorePools::<T>::mutate(asset, core_pool_id, |pool| {
+				let pool = pool.as_mut().ok_or(Error::<T>::BoostPoolDoesNotExist)?;
+				pool.add_funds(booster_id.clone(), amount);
+				Ok::<(), DispatchError>(())
+			})?;
+
+			Self::deposit_event(Event::<T>::BoostFundsAdded {
+				booster_id,
+				boost_pool: BoostPoolId { asset, tier: pool_tier },
+				amount,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::stop_boosting())]
+		pub fn stop_boosting(
+			origin: OriginFor<T>,
+			asset: Asset,
+			pool_tier: BoostPoolTier,
+		) -> DispatchResult {
+			let booster = T::AccountRoleRegistry::ensure_liquidity_provider(origin)?;
+			ensure!(T::SafeMode::get().stop_boosting_enabled, Error::<T>::StopBoostingDisabled);
+
+			let boost_pool: BoostPool =
+				BoostPools::<T>::get(asset, pool_tier).ok_or(Error::<T>::BoostPoolDoesNotExist)?;
+
+			let core_pool_id = boost_pool.core_pool_id;
+
+			let (unlocked_amount, pending_loans) =
+				CorePools::<T>::mutate(asset, core_pool_id, |pool| {
+					let pool = pool.as_mut().ok_or(Error::<T>::BoostPoolDoesNotExist)?;
+
+					pool.stop_lending(booster.clone()).map_err(|e| match e {
+						core_lending_pool::Error::AccountNotFoundInPool =>
+							Error::<T>::AccountNotFoundInBoostPool,
+					})
+				})?;
+
+			T::Balance::credit_account(&booster, asset, unlocked_amount);
+
+			// Map pending loans to pending boosts (for now we do it the expensive way, but
+			// if needed we could optimise this with a map loan_id -> boost_id):
+			let pending_boosts = {
+				BoostedDeposits::<T>::iter_prefix(asset)
+					.filter_map(|(deposit_id, pool_contributions)| {
+						pool_contributions
+							.values()
```

Comment:
Shouldn't this be `pool_contributions.get(pool_tier)`?

Reply:
You are right, though no longer relevant now that I change this to use `LoanUsage` as you suggested above.